interface-elements-for-web/articles/report-designer/bind-to-data/bind-a-report-to-data.md

```diff
@@ -0,0 +1,53 @@
+---
+title: Bind a Report to Data
+author: Anna Vekhina
+---
+# Bind a Report to Data
+
+This tutorial illustrates how to bind a report to a hierarchical data source and specify a master-detail relationship between data source queries.
+
+## Add a New Data Source
+
+1. [Create a new blank report](../../report-designer/add-new-reports.md).
+2. Invoke the designer [menu](../report-designer-tools/menu.md) and click **Add SQL Data Source...**.
+
+   ![](../../../images/eurd-web-add-sql-data-source-menu.png) 
+
+   The menu contains this command if your software vendor provides data connections.
+
+3. Select the data connection on the [Data Source Wizard](../report-designer-tools/data-source-wizard.md)'s first page and click **Next**.
+
+    ![](../../../images/eurd-web-binding-sql-ds-wizard.png)
+
+4. Choose which tables, views and/or stored procedures to add to a report and click **Next**. Select two or more tables to [create a master-detail report](../create-popular-reports/create-a-master-detail-report-use-detail-report-bands.md).
+
+    ![](../../../images/eurd-web-binding-sql-ds-wizard-tables.png)
+    
+    > [!NOTE]
+    > You can also create [custom queries](../report-designer-tools/report-wizard/data-bound-report/connect-to-a-database/create-a-query-or-select-a-stored-procedure.md) to shape data at the data source level. Select **Queries** and click the plus button to add a new query.
+    > 
+    > The invoked [Query Builder](../report-designer-tools/query-builder.md) allows you to join multiple tables, filter, sort, and group their data, and calculate various aggregate functions.
+    > 
+    > Although you can join different tables within a single query, master-detail data sources are preferred in most cases to provide better performance.
```

Comment:
Clarify. Why 'Although'?

Reply:
You can join different tables within a single query, but master-detail data sources are preferred in most cases to provide better performance.